internal/hardware/validator.go

```diff
@@ -147,25 +148,45 @@ func (v *validator) DiskIsEligible(ctx context.Context, disk *models.Disk, infra
 			fmt.Sprintf(wrongDriveTypeTemplate, disk.DriveType, strings.Join(v.getValidDeviceStorageTypes(hostArchitecture, clusterVersion), ", ")))
 	}
 
-	// We only allow multipath if all paths are FC
 	if disk.DriveType == models.DriveTypeMultipath {
+		fc, iscsi := false, false
```

Comment:
we can rely on `GetDisksOfHolder` : 
```
fc := GetDisksOfHolder(inventory.Disks, disk, models.DriveTypeFC)
iscsi := GetDisksOfHolder(inventory.Disks, disk, models.DriveTypeISCSI)
// then compare the len
// if len(fc) == 0 && len(iscsi) == 0 => wrongMultipathType
// if len(fc) > 0 && len(fc) != len(strings.Split(inventoryDisk.Holders, ",")) => mixedTypesInMultipath
// if len(iscsi) > 0 && len(iscsi) != len(strings.Split(inventoryDisk.Holders, ",")) => mixedTypesInMultipath
// for d in iscsi {... do check errors for each iscsi drive...}
```
it would reduce the level of imbrication (to makes the code more readable), and rely on the function created in the scope of this PR,